test/truncate.test.js

```diff
@@ -0,0 +1,170 @@
+import assert from 'assert'
+import lodashStable from 'lodash'
+import truncate from '../truncate.js'
+
+describe('truncate', () => {
+  const string = 'hi-diddly-ho there, neighborino'
+
+  describe('Control-Flow Graph structural tests', () => {
+    it('path::1::should return long string truncated with omission', () => {
+      const result = truncate('32-chars-string-----------------')
+
+      assert.strictEqual(result, '32-chars-string------------...')
+    })
+
+    it('path::2::should truncate strings with unicode', () => {
+      const result = truncate('32-chars-string\ud800----------------')
+
+      assert.strictEqual(result, '32-chars-string\ud800-----------...')
+    })
+
+    it('path::3::should not truncate if `string` is <= `length` default value (30)', () => {
+      const result = truncate('small string')
+
+      assert.strictEqual(result, 'small string')
+    })
+
+    it('path::4::should truncate to `omission` when it is larger than length', () => {
+      const result = truncate('32-chars-string-----------------', {
+        omission: 'a-omission-string-larger-than-the-length'
+      })
+
+      assert.strictEqual(result, 'a-omission-string-larger-than-the-length')
+    })
+
+    it('path::5::should not use `separator` if it is already at the end', () => {
+      const result = truncate('32-chars-string------------|----', {
+        separator: '|'
+      })
+
+      assert.strictEqual(result, '32-chars-string------------...')
+    })
+
+    it('path::6::should not use `separator` at the end with unicode chars', () => {
+      const result = truncate('32-chars-string-with-\ud800-----|----', {
+        separator: '|'
+      })
+
+      assert.strictEqual(result, '32-chars-string-with-\ud800-----...')
+    })
+
+    it('path::7::should not use `separator` when it is the truncated part', () => {
+      const result = truncate('32-chars-string---------------|-', {
+        separator: '|'
+      })
+
+      assert.strictEqual(result, '32-chars-string------------...')
+    })
+
+    it('path::8::should use last `separator` occurrence to slice string before length', () => {
+      const result = truncate('32-chars-string   --------------', {
+        separator: ' '
+      })
+
+      assert.strictEqual(result, '32-chars-string  ...')
+    })
+
+    it('path::9::should ignore `separator` if is a regex without matches', () => {
+      const result = truncate('32-chars-string-----------------', {
+        separator: /no-match-pattern/
+      })
+
+      assert.strictEqual(result, '32-chars-string------------...')
+    })
+
+    it('path::10::should use not use `separator` regex to slice string', () => {
```

Comment:
```suggestion
    it('should use not use `separator` regex to slice string', () => {
```

Reply:
'should use' or 'should not use'? hahaha

Reply:
It's a Schrodinger's test case. It should do both at the same time. Haha